### PR TITLE
chore(ui-react): update eslint config

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Remove `eslint-config-airbnb` and `eslint--config-airbnb-typescript` plugins and other `eslint` updates:
- remove aforementioned configs as:
  - the configs are very opinionated and come with a ton of predefined rules, creating more maintenance overhead in both resolving linting errors and determining which preconfigured rules need to be overridden
  - allows for easier refinement of applied `eslint` rules to match better serve the `ui-react` codebase, and ensures they are easily visible in the local _.eslintrc.js_ config
- remove `import` overrides related to the aforementioned configs
- add back some rules from the aforementioned configs
- improve file structure/commenting
- add `eslint:recommended` rule set
- add additional `react` rules
- replace `plugin:prettier/recommended` with just `prettier` to prevent extraneous errors while typing
- add _rollup.config.js_ and _jest.setup.ts_ to `ignorePatterns`

#### Issue #, if available
NA

#### Description of how you validated changes
Manually validated rule updates

#### Checklist

- [x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/.eslintrc.js [hunk 1]
```diff
@@ -54,70 +75,75 @@ module.exports = {
...
         'prefer-destructuring': ['error', { array: false, object: true }],
-        'prettier/prettier': ['error'],
+
+        // jest rules either not in recommended rule set or overridden
+        'jest/expect-expect': ['error', { assertFunctionNames: ['expect*'] }],
+        'jest/no-mocks-import': 'off',
+
+        // react rules either not in recommended rule set or overridden
+        'react/destructuring-assignment': ['error', 'always'],
+        'react/jsx-boolean-value': 'error',
```
Comment (reesscot) [inline packages/react/.eslintrc.js:134]: This doesn't appear to be the correct value for the rule?
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md

Comment (calebpollman) [reply packages/react/.eslintrc.js:134]: You would think that but:

image

In fact:
image

Comment (reesscot) [reply packages/react/.eslintrc.js:134]: lol, so I guess their docs are out of date.

**Review by reesscot — APPROVED** [review]
Shipit

**Review by zchenwei — APPROVED** [review]
:shipit:

**Pull request merged**